common/djangoapps/util/tests/test_request.py

```diff
@@ -50,8 +50,26 @@ def test_course_id_from_url(self):
         self.assertIsNone(course_id_from_url('/login'))
         self.assertIsNone(course_id_from_url('/course/edX/maths/2020'))
         self.assertIsNone(course_id_from_url('/courses/edX/maths/'))
+        self.assertIsNone(course_id_from_url('/api/courses/v1/blocks/edX/maths/2020'))
+        self.assertIsNone(course_id_from_url('/api/courses/v1/blocks/course-v1:incidental+courseid+formatting'))
+        self.assertIsNone(course_id_from_url('/api/courses/v41/notcourses/course-v1:incidental+courseid+formatting'))
 
         course_id = course_id_from_url('/courses/edX/maths/2020')
         self.assertEqual(course_id.org, 'edX')
         self.assertEqual(course_id.course, 'maths')
         self.assertEqual(course_id.run, '2020')
+
+        course_id = course_id_from_url('/courses/course-v1:edX+maths+2020')
+        self.assertEqual(course_id.org, 'edX')
+        self.assertEqual(course_id.course, 'maths')
+        self.assertEqual(course_id.run, '2020')
+
+        course_id = course_id_from_url('/api/courses/v1/courses/course-v1:edX+maths+2020')
+        self.assertEqual(course_id.org, 'edX')
+        self.assertEqual(course_id.course, 'maths')
+        self.assertEqual(course_id.run, '2020')
+
+        course_id = course_id_from_url('/api/courses/v1/courses/edX/maths/2020')
+        self.assertEqual(course_id.org, 'edX')
+        self.assertEqual(course_id.course, 'maths')
+        self.assertEqual(course_id.run, '2020')
```

Comment:
Better yet create a course key with the expected values, and do a single equality check, between the created course_id and the expected value.